api/swagger.json

```diff
@@ -0,0 +1,1304 @@
+      }
+    },
+    "/chapters/{chapterId}": {
+      "get": {
+        "tags": [
+          "chapter"
+        ],
+        "summary": "Retreive the details of a chapter",
+        "description": "",
+        "operationId": "findChapterById",
+        "parameters": [
+          {
+            "name": "chapterId",
+            "in": "path",
+            "description": "ID of chapter to return",
+            "required": true,
+            "schema": {
+              "type": "integer",
+              "format": "int64"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Chapter"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "Not authorized"
+          },
+          "400": {
+            "description": "Invalid ID supplied"
+          },
+          "404": {
+            "description": "Chapter not found"
+          },
+          "default": {
+            "description": "unexpected error",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/chapters/{chapterId}/events": {
+      "get": {
+        "tags": [
+          "chapter"
+        ],
+        "summary": "Retreive the events belonging to a chapter",
+        "description": "",
+        "operationId": "findEventsInChapter",
+        "parameters": [
+          {
+            "name": "chapterId",
+            "in": "path",
+            "description": "ID of chapter to return",
+            "required": true,
+            "schema": {
+              "type": "integer",
+              "format": "int64"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/Event"
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "Not authorized"
+          },
+          "400": {
+            "description": "Invalid ID supplied"
+          },
+          "404": {
+            "description": "Chapter not found"
+          },
+          "default": {
+            "description": "unexpected error",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/chapters/{chapterId}/events/{eventId}": {
```

Comment:
Probably makes more sense to pull this down to /events/{eventId}. Each event ID should be globally unique to the instance anyway (and probably in general, yay UUIDs!), and thus the event ID is all you really need to find it. That also avoids the temptation to keep stacking sub-resources on top, so you have even more IDs to keep track of to retrieve things.

By comparison, /chapters/{chapterId}/events (the subresource collection endpoint) makes perfect sense where it is.

Reply:
Agreed, but please make it `/event/{eventId}` because it'll return one event only.